### PR TITLE
Removed embedded autover version.py

### PR DESCRIPTION
**Pull request by jlstevens**
I'll now test this PR against the param 1.6 alpha versions.

### CODE REVIEW

**Comment by jlstevens** [comment]
The tests are failing right now, but that should only be because param 1.6.0 isn't actually released yet.

Locally (testing against the alpha) it all seems to be working! ``conda build .`` runs fine and when I install the resulting package, the version is reported correctly.

**Comment by jlstevens** [comment]
Using ``conda build -c pyviz .`` (as param 1.6 isn't on defaults yet) everything seems good.

@ceball @philippjfr Should we merge now or wait till param 1.6 is no defaults so we can rerun the tests and get them green?

**Comment by jlstevens** [comment]
The tests are green using the 1.6 release of param. I also tested conda build and it works fine. 

Ready to review/merge.

**Comment by jbednar** [comment]
Looks good to me, if it works!

File: setup.py [hunk 1]
```diff
@@ -56,11 +56,15 @@ def get_setup_version(reponame):
...
+    try:
+        from param import version
+    except:
+        embed_version(basepath)
+        version = importlib.import_module("version")
     if version is not None:
         return version.Version.setup_version(basepath, reponame, archive_commit="$Format:%h$")
     else:
-        print("WARNING: autover unavailable. If you are installing a package, this warning can safely be ignored. If you are creating a package or otherwise operating in a git repository, you should refer to autover's documentation to bundle autover or add it as a dependency.")
+        print("WARNING: param>=1.6.0 unavailable. If you are installing a package, this warning can safely be ignored. If you are creating a package or otherwise operating in a git repository, you should install param>=1.6.0.")
```
Comment (ceball) [inline setup.py]: I don't see how this warning will be reached. Either embed_version() worked and version's not None, or embed_version() fails...?

Comment (jlstevens) [reply setup.py]: Good point. Happy to remove it in another PR.

Comment (jlstevens) [reply setup.py]: I think that could be easily fixed by sticking in a try/except around embed version. i.e

```python
try:
   from param import version
except:
   version = embed_version(basepath) # Returns the module or None if failure
```

In this case, the warning is valid and it falls back to the JSON file. I'll make a PR for holoviews using this approach now...the key thing then will be to make sure this is used in other projects too (param doesn't need this but datashader probably does).

Comment (jlstevens) [reply setup.py]: I've made a PR to address this: #2532

**Comment by ah-** [comment]
Haven't looked in detail yet, but this seems to have broken pip install for me:

```
Collecting git+https://github.com/ioam/holoviews@fd6bac20c882fdbcafb1e788ed20e78a45585025
  Cloning https://github.com/ioam/holoviews (to fd6bac20c882fdbcafb1e788ed20e78a45585025) to /tmp/pip-d25zyy5f-build
  Could not find a tag or branch 'fd6bac20c882fdbcafb1e788ed20e78a45585025', assuming commit.
    Complete output from command python setup.py egg_info:
    WARNING: param>=1.6.0 unavailable. If you are installing a package, this warning can safely be ignored. If you are creating a package or otherwise operating in a git repository, you should install param>=1.6.0.
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-d25zyy5f-build/setup.py", line 75, in <module>
        version=get_setup_version("holoviews"),
      File "/tmp/pip-d25zyy5f-build/setup.py", line 71, in get_setup_version
        return json.load(open(version_file_path, 'r'))['version_string']
    FileNotFoundError: [Errno 2] No such file or directory: '/tmp/pip-d25zyy5f-build/holoviews/.version'
```

**Comment by jbednar** [comment]
Can you add the command line invocation you used?  In your case you can install param>=1.6.0 to fix the issue, but we'd like to know how this came up (i.e. why pip didn't already upgrade your copy of param).

**Comment by ah-** [comment]
Sure, it's this:
```
Step 12/27 : RUN pip install     tqdm     distributed==1.21.5     tornado==4.5     git+https://github.com/dask/dask-kubernetes@b850eb62de5937bf6ba32637bdadabf0bfc082fa     git+https://github.com/ioam/holoviews     git+https://github.com/jupyterhub/nbserverproxy     --no-cache-
dir     --upgrade
 ---> Running in 81e8e949ef8d
Collecting git+https://github.com/dask/dask-kubernetes@b850eb62de5937bf6ba32637bdadabf0bfc082fa
  Cloning https://github.com/dask/dask-kubernetes (to b850eb62de5937bf6ba32637bdadabf0bfc082fa) to /tmp/pip-ugx8ebjz-build
  Could not find a tag or branch 'b850eb62de5937bf6ba32637bdadabf0bfc082fa', assuming commit.
Collecting git+https://github.com/ioam/holoviews@fd6bac20c882fdbcafb1e788ed20e78a45585025
  Cloning https://github.com/ioam/holoviews (to fd6bac20c882fdbcafb1e788ed20e78a45585025) to /tmp/pip-d25zyy5f-build
  Could not find a tag or branch 'fd6bac20c882fdbcafb1e788ed20e78a45585025', assuming commit.
    Complete output from command python setup.py egg_info:
    WARNING: param>=1.6.0 unavailable. If you are installing a package, this warning can safely be ignored. If you are creating a package or otherwise operating in a git repository, you should install param>=1.6.0.
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-d25zyy5f-build/setup.py", line 75, in <module>
        version=get_setup_version("holoviews"),
      File "/tmp/pip-d25zyy5f-build/setup.py", line 71, in get_setup_version
        return json.load(open(version_file_path, 'r'))['version_string']
    FileNotFoundError: [Errno 2] No such file or directory: '/tmp/pip-d25zyy5f-build/holoviews/.version'
```

**Comment by jlstevens** [comment]
Thank you for reporting this @ah- !

I'm glad this was caught before the release, it was simply a missing import fixed in the above PR. Could you please try again while checking sure param isn't installed beforehand?

**Comment by ah-** [comment]
Just re-ran it with latest master and works perfectly!

**Comment by jlstevens** [comment]
Great! Thanks for getting back so quickly!

**Pull request merged**